File: src/Cognition.Console/src/types/events.ts
// Event DTOs for chat bus events

export interface AssistantMessageAppended {
  conversationId: string;
  agentId?: string;
  personaId?: string;
  content: string;
  timestamp: string;
  messageId?: string;
}

export interface PlanReady {
  conversationId: string;
  agentId: string;
  personaId: string;
  providerId: string;
  modelId?: string | null;
  plan: any;
  conversationPlanId: string;
  fictionPlanId: string;
  branchSlug: string;
  metadata?: Record<string, any> | null;
  timestamp: string;
}

export interface ToolExecutionRequested {
  conversationId: string;
  agentId: string;
  personaId: string;
  tool: string;
  args: Record<string, any>;
  conversationPlanId?: string | null;
  stepNumber?: number;
  fictionPlanId?: string | null;
  branchSlug?: string;
  metadata?: Record<string, any> | null;
  timestamp: string;
}

export interface ToolExecutionCompleted {
  conversationId: string;
  agentId: string;
  personaId: string;
  tool: string;
  result: any;
  success: boolean;
  error?: string;
  conversationPlanId?: string | null;
  stepNumber?: number;
  fictionPlanId?: string | null;
  branchSlug?: string;
  metadata?: Record<string, any> | null;
  timestamp: string;
}

// Streaming token delta during assistant generation
export interface AssistantTokenDelta {
  conversationId: string;
  agentId?: string;
  personaId?: string;
  // server may send delta/content/token; accept any
  delta?: string;
  content?: string;
  token?: string;
  timestamp?: string;
}

export interface ConversationCreated {
  conversationId: string;
  agentId?: string;
  personaId?: string;
  title?: string | null;
  timestamp?: string;
}

export interface ConversationJoined {
  conversationId: string;
  agentId?: string;
  personaId?: string;
  timestamp?: string;
}

export interface ConversationLeft {
  conversationId: string;
  agentId?: string;
  personaId?: string;
  timestamp?: string;
}

export interface ConversationUpdated {
  conversationId: string;
  title?: string | null;
  timestamp?: string;
}

export type ChatBusEvent =
  | AssistantMessageAppended
  | PlanReady
  | ToolExecutionRequested
  | ToolExecutionCompleted
  | AssistantTokenDelta
  | ConversationCreated
  | ConversationJoined
  | ConversationLeft
  | ConversationUpdated;

File: src/Cognition.Contracts/Events.cs
using System;
using System.Collections.Generic;

namespace Cognition.Contracts.Events
{
    public record UserMessageAppended(Guid ConversationId, Guid PersonaId, string Input);

    public record PlanRequested(
        Guid ConversationId,
        Guid AgentId,
        Guid PersonaId,
        Guid ProviderId,
        Guid? ModelId,
        string Input,
        int MinSteps,
        int MaxSteps,
        Guid FictionPlanId,
        string BranchSlug,
        Dictionary<string, object?>? Metadata = null);

    public record PlanReady(
        Guid ConversationId,
        Guid AgentId,
        Guid PersonaId,
        Guid ProviderId,
        Guid? ModelId,
        ToolPlan Plan,
        Guid ConversationPlanId,
        Guid FictionPlanId,
        string BranchSlug,
        Dictionary<string, object?>? Metadata = null);

    public record ToolExecutionRequested(
        Guid ConversationId,
        Guid AgentId,
        Guid PersonaId,
        string Tool,
        Dictionary<string, object?> Args,
        Guid? ConversationPlanId = null,
        int StepNumber = 0,
        Guid? FictionPlanId = null,
        string BranchSlug = "main",
        Dictionary<string, object?>? Metadata = null);

    public record ToolExecutionCompleted(
        Guid ConversationId,
        Guid AgentId,
        Guid PersonaId,
        string Tool,
        object? Result,
        bool Success,
        string? Error,
        Guid? ConversationPlanId = null,
        int StepNumber = 0,
        Guid? FictionPlanId = null,
        string BranchSlug = "main",
        Dictionary<string, object?>? Metadata = null);

    public record FictionPhaseProgressed(
        Guid PlanId,
        Guid ConversationId,
        Guid AgentId,
        string BranchSlug,
        string Phase,
        string Status,
        string? Summary,
        Dictionary<string, object?>? Payload,
        string? BacklogItemId = null);

    public record AssistantMessageAppended(Guid ConversationId, Guid PersonaId, string Content);

    public record ToolPlan(string Json);
}
